### PR TITLE
docs(swagger): set up annotation-based documentation

### PR DESCRIPTION
**Pull request by annibalsilva**
# Description

Remove the hand-made OpenAPI file and set the project to generate such file based on annotations on the code and publish it on the application itself.

The file can be seen on `/swagger-ui/index.html`.

Fixes [FSADT2-355](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-355)

## Type of change

Please delete options that are not relevant.

- [x] This change requires a documentation update

# How Has This Been Tested?

- [ ] Access `/swagger-ui/index.html` ([here](https://nrbestapi-63-service-api.apps.silver.devops.gov.bc.ca/swagger-ui/index.html)) and check if the documentation content and presentation is acceptable.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

### CODE REVIEW

File: src/main/java/ca/bc/gov/backendstartapi/endpoint/VegetationCodeEndpoint.java [hunk 1]
```diff
@@ -23,6 +31,7 @@
 @RestController
 @RequestMapping(path = "/api/vegetation-codes")
 @Validated
+@Tag(name = "vegetationCode", description = "Codes describing various vegetation species")
 public class VegetationCodeEndpoint {
 
   @Autowired private VegetationCodeRepository vegetationCodeRepository;
```
Comment (mishraomp) [inline src/main/java/ca/bc/gov/backendstartapi/endpoint/VegetationCodeEndpoint.java:37]: though not related to the swagger documentation changes, more in general, please avoid field injection.

File: src/main/java/ca/bc/gov/backendstartapi/endpoint/VegetationCodeEndpoint.java [hunk 2]
```diff
@@ -50,13 +72,32 @@ public VegetationCode findByCode(@PathVariable("code") String code) {
...
    *     PaginationParameters}
-   * @return a list of {@code size} or less vegetation codes, ordered by identifier
+   * @return a list of {@code perPage} or less vegetation codes matching {@code search}, ordered by
+   *     identifier
    */
   @GetMapping(produces = MimeTypeUtils.APPLICATION_JSON_VALUE)
+  @Operation(
+      summary = "Search for valid vegetation codes by their identifier or description",
+      description =
+          "Search for valid vegetation codes (ones which `effectiveDate` ≤ today < `expiryDate`) "
```
Comment (mishraomp) [inline src/main/java/ca/bc/gov/backendstartapi/endpoint/VegetationCodeEndpoint.java:82]: can we use `text blocks """` for multi line strings

**Review by mishraomp — APPROVED** [review]
LGTM, couple of minor comments, feel free to keep it in the backlog.

**Review by RMCampos — COMMENTED** [review]
LGTM. But I'm receiving a 500 error when trying a search request.

**Review by RMCampos — APPROVED** [review]
Great job!! 💯

**Pull request merged**